index.d.ts

```diff
@@ -1,13 +1,13 @@
-type Listener = (...args) => void
+type Listener<EventArgument> = (...args: EventArgument[]) => void
 type Unlisten = () => void
 
-export interface Emitter {
-	on(event: string, listener: Listener): Unlisten,
-	once(event: string, listener: Listener): Unlisten,
-	emit(event: string, ...args): void,
+export interface Emitter<EventName extends string, EventArgument> {
+	on(event: EventName, listener: Listener<EventArgument>): Unlisten,
+	once(event: EventName, listener: Listener<EventArgument>): Unlisten,
+	emit(event: EventName, ...args: EventArgument[]): void,
 	removeAllListeners(): void,
 }
 
-declare function makeEmitter<Object>(object?: Object): Emitter & Object
```

Comment:
I think TypeScript can handle multiple type definitions for one function - can we keep this old type around too, so that consumers aren't forced to specify all of the string values before using an emitter?

```ts
declare function makeEmitter<Object>(object?: Object): Emitter<string, any> & Object
```